### PR TITLE
Add ocp 4.8.2 cli support for ci against ocp 4.8

### PR DESCRIPTION
**Pull request by codihuston**
### What does this PR do?
- _What's changed? Why were these changes made?_

    Added cli support for testing against OCP 4.8 by implementing v4.8.2 for the cli. Using older 3.* cli threw exceptions
    when describing k8s resources in openshift.

- _How should the reviewer approach this PR, especially if manual tests are required?_

    - Review pipeline (attached to this PR)
    - Review [pipeline with TEST_OCP_NEXT enabled here](https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--secrets-provider-for-k8s/detail/fix%2Focp-next/2/pipeline/140)
    - Review changes to CHANGELOG

- _Are there relevant screenshots you can add to the PR description?_

    N/A

### What ticket does this PR close?
Resolves ONYX-11430

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or

    The test against the OCP 4.8 (next) cluster is not new, per se, but has not been passing until these changes.

- [ ] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- Support for OpenShift 4.8 has been certified as of this release.
```
Comment (imheresamir) [inline CHANGELOG.md:10]: Has it actually been "certified"? Are these changes enough to declare this? JW

File: deploy/test/Dockerfile [hunk 2]
```diff
@@ -21,8 +21,9 @@ RUN wget -O /usr/local/bin/kubectl ${KUBECTL_CLI_URL:-https://storage.googleapis
 ARG OPENSHIFT_CLI_URL
 RUN mkdir -p ocbin && \
     wget -O oc.tar.gz ${OPENSHIFT_CLI_URL:-https://github.com/openshift/origin/releases/download/v3.7.2/openshift-origin-client-tools-v3.7.2-282e43f-linux-64bit.tar.gz} && \
-    tar xvf oc.tar.gz --strip-components=1 -C ocbin && \
-    mv ocbin/oc /usr/local/bin/oc && \
+    tar xvf oc.tar.gz -C ocbin && \
+    ls -la ocbin && \
```
Comment (imheresamir) [inline deploy/test/Dockerfile:25]: Is this change needed?

Comment (codihuston) [reply deploy/test/Dockerfile:25]: OCP cli extracts differently between v3.* and v4.*. See: https://github.com/conjurdemos/kubernetes-conjur-demo/pull/144#discussion_r692377626

Comment (imheresamir) [reply deploy/test/Dockerfile:25]: Ah gotcha. This works too:
```
RUN mkdir -p ocbin && \
    wget -O oc.tar.gz ${OPENSHIFT_CLI_URL:-https://github.com/openshift/origin/releases/download/v3.7.2/openshift-origin-client-tools-v3.7.2-282e43f-linux-64bit.tar.gz} && \
    tar xvf oc.tar.gz --transform='s/.*\///' -C ocbin` && \
    mv ocbin/oc /usr/local/bin/oc && \
    rm -rf ocbin oc.tar.gz
```
Don't feel too strongly about it though. In any case the `ls` line can be removed

File: CHANGELOG.md (hunk 1 above)
Comment (codihuston) [reply CHANGELOG.md:10]: I may have jumped the gun here, but I was under the impression that this was already certified for 4.8, but cleaning up the tests was just a matter of ensuring we tested against 4.8. We may want to have someone clarify this.

File: deploy/test/Dockerfile (hunk 2 above)
Comment (codihuston) [reply deploy/test/Dockerfile]: I went ahead and removed the `ls` command. I'm going to leave the rest as is just because it is consistent with the other repo for now. Just let me know if you want me to change it to yours for sure and I can =).

**Comment by codihuston** [comment]
This PR stemmed from this JIRA task: [ONYX-11430](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-11430). Essentially, Infra wanted a test file that would test a set of projects against the OCP NEXT cluster.

File: CHANGELOG.md (hunk 1 above)
Comment (imheresamir) [reply CHANGELOG.md:10]: Yes please do - if we just started testing on 4.8 now then I'm not sure how it could have been compatibility "certified" before :)

**Review by imheresamir — APPROVED** [review]
LGTM

**Review by imheresamir — APPROVED** [review]
